R/connect.R

```diff
@@ -50,17 +50,17 @@ NULL
 #' }
 #' @export
 setMethod("dbConnect", "MySQLDriver",
-  function(drv, dbname = "", username = "", password = "", host = "",
-    unix.socket = "", port = 0, client.flag = 0,
-    groups = "rs-dbi", default.file = "", ...) {
+  function(drv, dbname = NULL, username = NULL, password = NULL, host = NULL,
+    unix.socket = NULL, port = 0, client.flag = 0,
+    groups = "rs-dbi", default.file = NULL, ...) {
 
     ptr <- connection_create(host, username, password, dbname, port, unix.socket,
       client.flag, groups, default.file)
 
     con <- new("MySQLConnection",
       ptr = ptr,
-      host = host,
-      db = dbname
+      host = if(is.null(host)) "" else host,
```

Comment:
Inspired by one of your examples from ['The S4 object system'], perhaps `NA_character_` is a better way to represent `NULL`. I added another commit for this (fb932a7).

Possible alternative: [Class union] ?
['The S4 object system']: http://adv-r.had.co.nz/S4.html
[Class union]: http://stackoverflow.com/questions/13002200/s4-classes-multiple-types-per-slot